agent/protocol/nats/nats.go

```diff
@@ -0,0 +1,771 @@
+	return fmt.Sprintf("Protocol:INFO,ServerID:%v,ServerName:%v,Version:%v,GoVersion:%v,Host:%v,Port:%v,MaxPayload:%v,TLSRequired:%v",
+		m.ServerID, m.ServerName, m.Version, m.GoVersion, m.Host, m.Port, m.MaxPayload, m.TLSRequired)
+}
+func (m *Info) ParseData(payload []byte) (*Info, int) {
+	// INFO {"option_name":option_value,...}␍␊
+	packetLen := readLine(payload)
+	if packetLen < 0 {
+		return nil, -1
+	}
+	parts := splitFields(payload[:packetLen-len(_CRLF_)])
+	if len(parts) != 2 {
+		return nil, packetLen
+	}
+	// if strings.ToUpper(string(parts[0])) != "INFO" {
+	// 	return nil, packetLen
+	// }
+
+	msg := Info{}
+	err := json.Unmarshal(parts[1], &msg)
+	if err != nil {
+		return nil, packetLen
+	}
+
+	msg.ProtocolCode = INFO
+	msg.isReq = false
+	msg.Buf = payload[:packetLen]
+	return &msg, packetLen
+}
+func (m *Info) Parse(payload []byte, messageType protocol.MessageType) (*NatsMessage, int) {
+	common.ProtocolParserLog.Debugf("NATS Parse Info:%d, %x", len(payload), string(payload))
+	msg, prasedLen := m.ParseData(payload)
+	if prasedLen < 0 || msg == nil {
+		return nil, prasedLen
+	}
+	common.ProtocolParserLog.Debugf("NATS Parsed Info:[%v], ReadBytes:%d", msg.String(), prasedLen)
+	return &msg.NatsMessage, prasedLen
+}
+
+// Connect
+func (m *Connect) String() string {
+	return fmt.Sprintf("Protocol:CONNECT,Verbose:%v,Pedantic:%v,TLSRequired:%v,Name:%v,Version:%v",
+		m.Verbose, m.Pedantic, m.TLSRequired, m.Name, m.Version)
+}
+
+func (m *Connect) ParseData(payload []byte) (*Connect, int) {
+	// CONNECT {"option_name":option_value,...}␍␊
+	packetLen := readLine(payload)
+	if packetLen < 0 {
+		return nil, -1
+	}
+	parts := splitFields(payload[:packetLen-len(_CRLF_)])
+	if len(parts) != 2 {
+		return nil, packetLen
+	}
+	// if strings.ToUpper(string(parts[0])) != "CONNECT" {
+	// 	return nil, packetLen
+	// }
+
+	msg := Connect{}
+	err := json.Unmarshal(parts[1], &msg)
+	if err != nil {
+		return nil, packetLen
+	}
+
+	msg.ProtocolCode = CONNECT
+	msg.isReq = true
+	msg.Buf = payload[:packetLen]
+	return &msg, packetLen
+}
+
+func (m *Connect) Parse(payload []byte, messageType protocol.MessageType) (*NatsMessage, int) {
+	common.ProtocolParserLog.Debugf("NATS Parse Connect:%d, %x", len(payload), string(payload))
+	msg, prasedLen := m.ParseData(payload)
+	if prasedLen < 0 || msg == nil {
+		return nil, prasedLen
+	}
+	common.ProtocolParserLog.Debugf("NATS Parsed Connect:[%v], ReadBytes:%d", msg.String(), prasedLen)
+	return &msg.NatsMessage, prasedLen
+}
+
+// Pub
+func (m *Pub) String() string {
+	return fmt.Sprintf("Protocol:PUB,Subject:%v,ReplyTo:%v,PayloadSize:%d,Payload:%x", m.Subject, m.ReplyTo, m.PayloadSize, m.Payload)
+}
+
+func (m *Pub) ParseData(payload []byte) (*Pub, int) {
+	// PUB <subject> [reply-to] <#bytes>␍␊[payload]␍␊
+	packetLen := readLine(payload)
+	if packetLen < 0 {
+		return nil, -1
+	}
+	parts := splitFields(payload[:packetLen-len(_CRLF_)])
+	if len(parts) < 3 || len(parts) > 4 {
```

Comment:
So either 3 or 4, right?

```suggestion
	switch len(parts) {
		case 3, 4:
		default:
```